### ISSUE TITLE
BackgroundAudio – Can’t change track or pause in the task bar

### ISSUE THREAD

**Issue by pekspro** [Microsoft/Windows-universal-samples#100]
The BackgroundAudio is pretty nice :-). But I miss some sample how to enable the controls to change track and pause in the taskbar when you hovering the icon. Currently they are just disabled. I thought this would be solved by using SystemMediaTransportControls which the sample use. But it seems to just work with the controls on the keyboard, not the task bar.

**Comment by aarononeal** [comment]
Glad you're enjoying the sample! Sorry about the taskbar controls. This is an OS bug we hope to address in a future update.

**Comment by pekspro** [comment]
Oh, I wasn’t expected that :-). Keep up the good work :-).

**Comment by pekspro** [comment]
I just tried this in the latest version with the latest SDK (10586) and the problem remains :-(. Any news about this?

**Comment by aarononeal** [comment]
This change must be made to the OS and not the SDK. We are still tracking it.

You can work around the problem by implementing the SystemMediaTransportControl in both the foreground process (via the GetForCurrentView() instance) and the background (via the media player instance).

**Comment by pekspro** [comment]
Thanks for quick answering :-). I was able to work around at by using SystemMediaTransportControls as you suggested. I found some funny things when I did that. Here some tips if someone else want to solve the same problem.

First I found out that it was important to call SystemMediaTransportControls.GetForCurrentView() before my background thread was started (that also created an instance of this object). It was also necessary to enable this control and the play/pause buttons. If they were disabled, it was for some reason not possible to enable them again (at least they were disabled in the taskbar).

I also found out that I needed to be careful with the SystemMediaTransportControls.DisplayUpdater object. If I used this object (maybe a call to DisplayUpdate.Update() was enough – I’m not sure) while the volume control with media information was visible (the control you see in Windows if you press volume up/down on the keyboard), my application would freeze next time a tried to access SystemMediaTransportControls.DisplayUpdater. So I simply decided to not use DisplayUpdater at all.

I also used this workaround in desktop mode only. In phone everything seems to be OK.

Even if this is not a perfect solution I’m quite happy with it :-).
